docs/design/coreclr/botr/logging.md

```diff
@@ -0,0 +1,70 @@
+
+*NOTE: This document will repeatedly reference many environment variables. For a detailed list of environment variables supported by the runtime and explanations of what they do, consult [`clrconfigvalues.h`](https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/clrconfigvalues.h), [`gcconfig.h`](https://github.com/dotnet/runtime/blob/main/src/coreclr/gc/gcconfig.h), and [`jitconfigvalues.h`](https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/jitconfigvalues.h).*
+
+Types of logging
+================
+
+ETW / EventPipe logging
+-----------------------
+"Event Tracing for Windows", or ETW, is an extensible tracing system built into Windows that various applications and services can use to report events in a way that allows the end user to filter and record them dynamically. ETW is consumed by tools like perfview, xperf, UIforETW, or WPR/WPA. This system is frequently used by customers to understand what is happening in development or production environments.
+
+EventPipe is a cross-platform tracing system similar to ETW that is used widely in the .NET runtime.
+For a detailed overview of EventPipe, see [the EventPipe documentation pages on .NET Learn](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/eventpipe). It offers excellent performance along with lots of flexibility, so it is one of the main ways to get information from an active runtime that we advertise. Most events that we support in debug builds also are exposed in release builds.
+
+For basic scenarios, you can use the [`dotnet-counters`](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-counters) tool to monitor events and performance counters the runtime reports through EventPipe. For example, to collect the default counters from a project while running it, you could use `dotnet-counters collect -- dotnet run`.
+
+You can also use the [`dotnet-trace`](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-trace) tool to capture EventPipe events from the runtime on the fly, either saving them to a file or sending them to your console. It provides command line options you can use to filter events by severity level or based on keywords. For example, to capture high-severity GC events only from a project while running it, you could use `dotnet-trace collect --clreventlevel error --clrevents gc -- dotnet run`.
+
+Please see the linked documentation pages for more information on how to use each of the above tools. They have an extensive set of options and some helpful features like time limits, attaching to existing processes, and configurable output formatting.
+
+For advanced scenarios, on Windows you can also use standard ETW tools like [WPR](https://learn.microsoft.com/en-us/windows-hardware/test/wpt/windows-performance-recorder) to capture events from EventPipe, and on Linux you can use [LTTNG](https://lttng.org/).
```

Comment:
For Linux, it may be work mentioning perfcollect script: https://learn.microsoft.com/en-us/dotnet/core/diagnostics/trace-perfcollect-lttng. It approximates the experience that you will get from system-wide ETW collection on Windows.